### PR TITLE
[§] cleans snippets interface and adds subgraph tests

### PR DESCRIPTION
**Pull request by geexie**
cleans up snippets interface according to it's architecture [guide](https://github.com/openvinotoolkit/openvino/wiki/SS)

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

File: inference-engine/src/snippets/src/pass/collapse_subgraph.cpp [hunk 1]
```diff
@@ -503,6 +510,7 @@ ngraph::snippets::pass::AttachToSubgraph::AttachToSubgraph(bool tokenize_by_node
                     << " inputs and " << subgraph->outputs().size()
                     << " outputs and " << subgraph->get_body()->get_ops().size() << " ops total\n";
 
+        // subgraph->serialize();
```
Comment (ilyachur) [inline inference-engine/src/snippets/src/pass/collapse_subgraph.cpp:513]: Looks like unnecessary code.

File: inference-engine/src/snippets/src/pass/collapse_subgraph.cpp [hunk 2]
```diff
@@ -189,17 +184,29 @@ auto has_supported_in_out(std::shared_ptr<Node> n) -> bool {
         if (in.get_tensor().get_element_type() != ngraph::element::f32) {
             return false;
         }
+
+        if (in.get_partial_shape().is_static() && in.get_shape().size() > 6) {
```
Comment (ilyachur) [inline inference-engine/src/snippets/src/pass/collapse_subgraph.cpp:188]: What if shapes are `dynamic`? Do we support such case?

Comment (geexie) [reply inference-engine/src/snippets/src/pass/collapse_subgraph.cpp:188]: SnippetS are not tested on dynamic shapes right now so I assume no support at all

Comment (geexie) [reply inference-engine/src/snippets/src/pass/collapse_subgraph.cpp:192]: done

**Pull request merged**